### ISSUE TITLE
Translate code (comments) to english

### ISSUE THREAD

**Issue by MaximumWoahverdrive** [HerissonMignion/FactorioOrganizer#4]
Right now many of the comments inside the code are in french. 
Translating these and adding some more where appropriate would definitely help

**Comment by HerissonMignion** [comment]
I'll work on that as soon as i'll be confident of what i do on github and that i'll not harass everybody with commit notification. Same for testing code :)

**Comment by MaximumWoahverdrive** [comment]
Dont worry about commit notifications, only those who have set the repo to watch everything will be notified of that and they have also chosen to be notified of every message on every issue, etc. 
Most of the "watching" people have likely set it to releases only.

**Comment by MaximumWoahverdrive** [comment]
I'd also be happy to help with anything if you need any help or have questions (regarding managing a github repo etc.)

**Comment by HerissonMignion** [comment]
I think i'd have a lot of question in a wide range but right now i'm eating :) i just need like 20 min. I uploaded "TestRepo" on github. i removed every file of my computer and i'll pratice github on it after eating. :)

**Comment by HerissonMignion** [comment]
I know that comments like that don't has to be in "translate code issue", where should we put that ?

**Comment by MaximumWoahverdrive** [comment]
Conversations like this dont really have a place in github as they are meant to be carried out on other platforms. If you want you can add me on discord or so, so we can chat better.

**Comment by HerissonMignion** [comment]
Don't have discord

**Comment by MaximumWoahverdrive** [comment]
What chat platform would you like to use? I just suggested discord because its the main thing i use

**Comment by HerissonMignion** [comment]
![ToDelete173](https://user-images.githubusercontent.com/38510629/63201609-02e72780-c054-11e9-9dad-3a2327c4071c.PNG)

So here's my first question. When i clone a repo in the vs extension, in the Change tab (up left of the grid of tab) i have ~70000 change because it consider my whole user folder. It's obvious it's not normal. Am i doing it wrong ? I can go double click the sln file and after it works normaly.

**Comment by MaximumWoahverdrive** [comment]
it looks like you have your .sln file in `C:\Users\Charles`. Because of this (or some other reason) it assumes that entire folder is the project. 

When cloning a Repo you should just go to "File" -> "Clone or check out code" and then enter the url (for this repo it would be `https://github.com/HerissonMignion/FactorioOrganizer.git`)
and specify a path to store it in. (this should be an empty folder. by default this would be `C:\Users\Charles\Source\Repos\FactorioOrganizer`)

**Comment by HerissonMignion** [comment]
I swear i have no sln file in my user folder. The sln is in C:\Users\Charles\source\repos\HerissonMignion\TestRepo\blabla.sln

**Comment by HerissonMignion** [comment]
Second question. When we merge two branch, how the merging of files work ? I guess that it overwrites the originale files, but are there special things i should know about that? Also (3e) in which occasion we normaly create new branch, when adding new fonctionnality, begin modification of any file for futur fonctionnality, something else ?

**Comment by MaximumWoahverdrive** [comment]
merging of files is automatically done by checking what changes were made not just what files were changed but also which lines were changed in those files. 
Say we create  a new branch at commit 10 then we make changes to the beginning of a file in the original branch. and in the new branch we make changes to the ending of that same file.
If we then try to merge the new branch into the original one, git automatically checks if the changes "overlap" and if they dont you can just click merge and the new file will have both changes applied without problems. 
if however the changes do conflict, for example because we made different changes to a method in both branches, we must decide which version from which branch we will keep in the merge.

Generally if github reports that a merge can be performed without conflict, it is safe to merge.

On the topic of when to create a new branch: that is really up to how you want to do it. Generally there are three approaches. (there are a few more but they dont really apply to a project this small)
1: Keep most changes in the master. (generally not recommended)
Not creating additional branches gets messy quickly if more than 1 or 2 people work on a project and failed or abandoned features quickly clutter the branch with junk commits, etc

2: Every new feature gets a branch (also not really recommended)
Keeps the main branch clean and tidy, but risks features conflicting with each other as they get merged to the master, often resulting in many manual merges of files and patchy code.

3: Single Dev branch (Probably the best)
All work is done in a dev branch. Whenever a release is ready, the dev branch gets merged into the master and a Release Tag is Created. This keeps the master clean while also not causing merging problems. 

If you were to do major changes like for example a big rewrite or something it would still make sense to create a new branch but generally the single dev branch is enough. Especially because when people create a fork to work on the project it is very similar to them creating a branch for themself to work in which will be merged back in.

So to sum up this workflow a bit:
Releases are done in master
Owner (you) and optionally Collaborators (these can: accept pull request, close and modify issues, etc) work in the dev branch
other people work in their forks of the repo
When either a new feature or imrovement (basically anything thats finished) is done you merge that into the master

**Comment by MaximumWoahverdrive** [comment]
To fix your issue It's probably easiest to just Clone the project again making sure the path is correct

**Comment by HerissonMignion** [comment]
I fixed the issue by double click the sln file.

**Comment by HerissonMignion** [comment]
THANKS A LOT

**Comment by MaximumWoahverdrive** [comment]
If you want me to help and if you trust me enough you can add me (or anyone else) as a collaborator to this repo.  
[What Collaborators can do](https://help.github.com/en/articles/permission-levels-for-a-user-account-repository#collaborator-access-on-a-repository-owned-by-a-user-account)  
[How to add a Collaborator](https://help.github.com/en/articles/inviting-collaborators-to-a-personal-repository)

**Comment by MaximumWoahverdrive** [comment]
Other than that you seem to know the basics. Again, feel free to ask any questions. (tho i would recommend getting some online chat platform like discord so you dont have to clutter issues in here with relatively random chatter)

**Comment by HerissonMignion** [comment]
So should i create a branch for myself as soon as possible (I'm already working on the actual issues, but on the master) ? If you're not a collaborator, can you merge stuff to master ?

**Comment by HerissonMignion** [comment]
Should i create a fork for myself instead of a brench ? I think i'm almost done of question. If you want me to get a discord to ask questions then i'll get one. Is discord the general used thing to "talk" on github ?

**Comment by MaximumWoahverdrive** [comment]
It's generally smart to work on a dev branch and if you dont have any uncommited changes now would be a good time to create/move to that branch. Its not urgent but it'll make sure people forking the project wont have too much stuff you are currently working on in there. 
If you're not a collaborator you have to make a pull request to merge stuff. 
Every merge starts with a pull request, collaborators can just also accept those requests themselves.
But they can also close issues, add labels to issues, create releases, and generally moderate/manage.
Collaborators are very similar to what would otherwise be called moderator.

**Comment by MaximumWoahverdrive** [comment]
It really varies. Discord is very popular but it really just comes down to what you platform you like to use. stuff like Rocket.Chat, Telegram, Whatsapp, Slack, etc work just as well.
There isnt really a general thing used to talk on github.

**Comment by MaximumWoahverdrive** [comment]
You cant actually create a fork because this is your Repo. 
So you would have to create a branch to work in if you dont want to do WIP stuff in the master

**Comment by HerissonMignion** [comment]
Thanks again !  You just made the life of someone (me) easier :)

**Comment by HerissonMignion** [comment]
It seems logic to me to finish comment translation and test code removing inside the master.

**Comment by HerissonMignion** [comment]
If you need more comments tell me here where you need comments. If that's enough, close the issue.
